gobblin-cluster/src/main/java/org/apache/gobblin/cluster/HelixUtils.java

```diff
@@ -359,13 +369,16 @@ static void handleJobTimeout(String workFlowName, String jobName, HelixManager h
   private static void deleteStoppedHelixJob(HelixManager helixManager, String workFlowName, String jobName)
       throws InterruptedException {
     WorkflowContext workflowContext = TaskDriver.getWorkflowContext(helixManager, workFlowName);
-    while (workflowContext.getJobState(TaskUtil.getNamespacedJobName(workFlowName, jobName)) != STOPPED) {
+    while (workflowContext != null &&
+        workflowContext.getJobState(TaskUtil.getNamespacedJobName(workFlowName, jobName)) != STOPPED) {
       log.info("Waiting for job {} to stop...", jobName);
       workflowContext = TaskDriver.getWorkflowContext(helixManager, workFlowName);
       Thread.sleep(1000);
     }
-    // deleting the entire workflow, as one workflow contains only one job
-    new TaskDriver(helixManager).deleteAndWaitForCompletion(workFlowName, 10000L);
+    if (workflowContext != null) {
+      // deleting the entire workflow, as one workflow contains only one job
+      new TaskDriver(helixManager).deleteAndWaitForCompletion(workFlowName, 10000L);
```

Comment:
I am okay with hard coding this timeout value to 10 seconds since the deletion has typically happened very fast. But can we have a separate variable and corresponding comment that explains that 10 seconds should be plenty of time and we should expect deletes to happen a few seconds?

Just to avoid magic numbers and better readability for future readers